commands/pokemon.py

```diff
@@ -129,24 +142,141 @@ def pause(self):
+                discord_windows = list(filter(lambda w: 'Discord | #' in w.title, all_windows))
+
+                # Get the first Discord window
+                window = discord_windows[0] if discord_windows else None
+
+                if window is None:
+                    print("No Discord window found")
+                    print(f'Caught a {pokemon_info["Rarity"]} Pokemon: {pokemon_name}')
+                    return
+
+                # Connect to the window using pywinauto
+                app = Application().connect(handle=window._hWnd)
+
+                # Connect to the window using pywinauto
+                app = Application().connect(handle=window._hWnd)
+
+                # Bring the window to the foreground
+                app.top_window().set_focus()
+
+                # Wait for a moment to let the window come to the foreground
+                time.sleep(1)
+
+                # Get the window's location
+                x, y, width, height = window.left, window.top, window.width, window.height
+
+                # Get the current time
+                now = datetime.now()
+
+                # Format the time in 12-hour format
+                time_string = now.strftime("%I_%M_%S_%p")
+
+                # Use time_string in your string
+                screenshot_path = f'screenshots/{pokemon_name}_{time_string}.png'
+                
+                # Calculate the center of the window
+                center_x = x + width // 2
+                center_y = y + height // 2
+
+                # Move the cursor to the center of the window
+                moveTo(center_x, center_y)
+
+                # Ensure the directory exists
+                os.makedirs(os.path.dirname(screenshot_path), exist_ok=True)
+
+                # Take a screenshot of the window
+                with mss.mss() as sct:
+                    screenshot = sct.grab({"top": y, "left": x, "width": width, "height": height})
+                    png_data = mss.tools.to_png(screenshot.rgb, screenshot.size)
+                    # Ensure the directory exists
+                    os.makedirs(os.path.dirname(screenshot_path), exist_ok=True)
+
+                # Write the PNG data to a file
+                with open(screenshot_path, 'wb') as f:
+                    f.write(png_data)
+
+                # Log the screenshot path
+                logger.info(f'{Fore.YELLOW}Screenshot taken of{Style.RESET_ALL} {Fore.GREEN}{pokemon_name}{Style.RESET_ALL} {Fore.YELLOW}and saved as {screenshot_path}{Style.RESET_ALL}')
+
+            # Choose the log color based on the Pokemon rarity
+            if pokemon_info["Rarity"] == "Legendary":
+                log_color = Fore.MAGENTA
+            elif pokemon_info["Rarity"] == "Super Rare":
+                log_color = Fore.CYAN
+            else:
+                log_color = Fore.YELLOW if pokemon_info.get("Shiny", False) else Fore.WHITE
```

Comment:
we can convert this if statement into a function so we can reuse in fishing.py and even in eggs.py